osbs/api.py

```diff
@@ -487,60 +469,78 @@ def create_build(self, **kwargs):
+        Take input args, create source pipeline run
 
-        :return: instance of BuildRequest
+        :return: instance of PiplelineRun
         """
+        error_messages = []
+        # most likely can be removed, source build should get component name
+        # from binary build OSBS2 TBD
+        if not component:
+            error_messages.append("required argument 'component' can't be empty")
+        if error_messages:
+            raise OsbsValidationException(", ".join(error_messages))
+
+        pipeline_run_name, pipeline_run_data = self._get_source_container_data()
+
         build_json_store = self.os_conf.get_build_json_store()
         user_params = SourceContainerUserParams.make_params(
             build_json_dir=build_json_store,
-            build_conf=self.build_conf,
+            build_conf=self.os_conf,
             component=component,
             koji_target=target,
             koji_task_id=koji_task_id,
+            pipeline_run_name=pipeline_run_name,
             **kwargs
         )
-        build_request = self.get_source_container_build_request(
-            outer_template=outer_template or ORCHESTRATOR_SOURCES_OUTER_TEMPLATE,
-            user_params=user_params
-        )
-        build_request.set_openshift_required_version(self.os_conf.get_openshift_required_version())
 
-        error_messages = []
-        if not component:
-            error_messages.append("required argument 'component' can't be empty")
-        if error_messages:
-            raise OsbsValidationException(", ".join(error_messages))
+        self._set_source_container_data(pipeline_run_name, pipeline_run_data, user_params)
 
-        builds_for_koji_task = []
-        if koji_task_id:
-            # try to find build for koji_task which isn't canceled and use that one
-            builds_for_koji_task = self._get_not_cancelled_builds_for_koji_task(koji_task_id)
+        logger.info("creating source container image pipeline run: %s", pipeline_run_name)
 
-        builds_count = len(builds_for_koji_task)
-        if builds_count == 1:
-            logger.info("found running build for koji task: %s",
-                        builds_for_koji_task[0].get_build_name())
-            response =\
-                BuildResponse(self.os.get_build(builds_for_koji_task[0].get_build_name()).json(),
-                              self)
-        elif builds_count > 1:
-            raise OsbsException("Multiple builds %s for koji task id %s" %
-                                (builds_count, koji_task_id))
-        else:
-            logger.info("creating source container image build")
-            response = self._create_build_directly(build_request)
+        pipeline_run = PipelineRun(self.os, pipeline_run_name, pipeline_run_data)
 
-        logger.debug(response.json)
-        return response
+        logger.info("pipeline run created: %s", pipeline_run.start_pipeline_run().json())
```

Comment:
Do we want to output the whole json here?

Reply:
there isn't any reason why not, we can always change it in the future

Reply:
it is only prun definifinition ofc